aip/auth/4110.md

```diff
@@ -0,0 +1,161 @@
+in an effortless manner.
+
+Auth libraries following the standards in these AIPs are known as _"Google
+Unified Auth Clients"_, or _GUAC_ for short. The resulting libraries are
+colloquially called _GUACs_.
+
+**Note:** Because this AIP describes guidance and requirements in a
+language-neutral way, it uses generic terminology which may be imprecise or
+inappropriate in certain languages or environments.
+
+## Guidance
+
+### Credential Types
+
+This section outlines the supported credential types of the ADC.
+
+- **GCloud CLI Credential**: A credential provided by the [GCloud CLI][0] that
+identifies a human user that needs to authenticate to access Google APIs. The
+auth libraries **must** support this credential type.
+
+- **Service Account Key**: A credential that identifies a non-human user that
+needs to authenticate to aceess Google APIs. The auth libraries **must** support
+this credential type.
+
+- **OAuth Client ID**: A credential that identifies the client application which
+allows human users to sign-in through [3-legged OAuth flow][1], which grants the
+permissions to the application to access Google APIs on behave of the human
+user. The auth libraries **may** support this credential type.
+
+### Environment Variables
+
+The auth libraries **must** support the following environment variables to allow
+developers to provide authentication configuration for their application:
+
+- **GOOGLE_APPLICATION_CREDENTIALS**: The specified value will be used as the
+full path for ADC to locate the credentials file. The credentials file
+**should** be one of the following types:
+
+    - Service account key
+    - GCloud CLI credentials
+
+    The credentials **may** be the OAuth Client ID if it is supported by the
+    auth library. This value __must__ be overridden by any user specified
+    credentials at the programming level (e.g. through clien-level options)
+
+- **GOOGLE_API_KEY**: [API keys][2] are a simple encrypted string that
+identifies an application, which are useful for accessing public data
+anonymously. The specified value of this environment variable **should** be used
+as the API key for the API requests. This value **must** be overridden by any
+user specified credentials at the programming level (e.g. through client
+options). And it **must** be ignored if GOOGLE_APPLICATION_CREDENTIALS is
+specified.
+
+- **GOOGLE_API_USE_CLIENT_CERTIFICATE:** The specified value **must** be
+either true or false. The client certificate **must** be ignored if this
+variable is set to false.
+
+### Inputs & Outputs
+
+From the input/output perspective, the inputs of _ADC_ __should__ be the
+credentials as well as the underlying environment that provides these
+credentials. 
+
+For example, the _GOOGLE_APPLICATION_CREDENTIALS_ environment variable can provide
+the default credential JSON as the input here, or the well-known path that
+gCloud uses to store the default user credential JSON. The output is the access
```

Comment:
Is the correct form "gCloud" or "gcloud"? I thought it was the latter. (@AlanGasperini is the authority here.)

Reply:
Let's call it "gcloud CLI"

https://cloud.google.com/sdk#section-5